node/messages/src/consensus_id.rs

```diff
@@ -32,15 +30,65 @@ impl MessageTrait for Box<ConsensusId> {
     }
 
     fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
-        writer.write_all(self.public_key.as_bytes())?;
-        writer.write_all(self.signature.as_bytes())?;
+        // bincode::serialize_into(writer, &(&self.public_key, &self.signature))?;
+        serde_json::to_writer(writer.by_ref(), &(&self.public_key, &self.signature))?;
 
         Ok(())
     }
 
     fn deserialize(bytes: BytesMut) -> Result<Self> {
-        let (public_key, signature) = bincode::deserialize_from(&mut bytes.reader())?;
+        let mut reader = bytes.reader();
+        // let (public_key, signature) = bincode::deserialize_from(&mut reader.by_ref())?;
+        let (public_key, signature) = serde_json::from_reader(&mut reader.by_ref())?;
 
         Ok(Box::new(ConsensusId { public_key, signature }))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use bytes::BufMut;
+    use narwhal_crypto::KeyPair;
+
+    use super::*;
+
+    #[test]
+    fn test_consensus_id() {
+        let mut rng = rand::thread_rng();
+        let keypair = KeyPair::new(&mut rng).unwrap();
+        let public = keypair.public();
+        println!("public: {:?}", public.to_bytes_le().unwrap().len());
+        let private = keypair.private();
+
+        let message = &[0u8; 32];
+        let signature = private.sign_bytes(message, &mut rng).unwrap();
+        println!("signature: {:?}", signature.to_bytes_le().unwrap().len());
+
+        let id = Box::new(ConsensusId { public_key: public.clone(), signature });
+        let mut buf = BytesMut::with_capacity(128).writer();
+        id.serialize(&mut buf).unwrap();
+        let bytes = buf.into_inner();
+        let deserialized = MessageTrait::deserialize(bytes).unwrap();
+        assert_eq!(id, deserialized);
+    }
+
+    #[test]
+    fn test_signature() {
```

Comment:
```suggestion
    fn signature_serialization() {
```

Reply:
also, the right place for serialization tests of snarkVM objects is snarkVM, so these tests should ultimately be removed if they exist there